### PR TITLE
[MFT] Fixes for forward track dataformat and track fitting 

### PR DESCRIPTION
**Pull request by rpezzi**
This PR: 
- Fixes a bug on forward tracks covariance matrix propagation;
- Adds optimized extrapolation function: parameters extrapolated as helix, errors extrapolated using quadratic approximation.
- MCS effects: 
  - Improved calculation of MCS effects
  - Layer-by-layer track propagation during track fitting for granular evaluation of MCS effects
- Robust seed determination for Kalman filter
- Minor adjustment on MFT geometry

### CODE REVIEW

**Review by bovulpes — DISMISSED** [review]

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h [hunk 1]
```diff
@@ -66,7 +67,7 @@ class TrackParFwd
   Double_t getInvPx() const { return 1. / getPx(); }                // return invpx
 
   Double_t getPy() const { return TMath::Sin(getPhi()) * getPt(); } // return py
-  Double_t getInvPy() const { return 1. / getPx(); }                // return invpy
+  Double_t getInvPy() const { return 1. / getPy(); }                // return invpy
```
Comment (shahor02) [inline DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h]: I don't see where these `getInvP...` are used, but they are prone to create a division by 0

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h [hunk 2]
```diff
@@ -126,11 +127,11 @@ class TrackParCovFwd : public TrackParFwd
   TrackParCovFwd() = default;
   ~TrackParCovFwd() = default;
   TrackParCovFwd& operator=(const TrackParCovFwd& tpf) = default;
-  TrackParCovFwd(const Double_t z, const SMatrix5 parameters, const SMatrix55 covariances, const Double_t chi2);
+  TrackParCovFwd(const Double_t z, const SMatrix5 parameters, const SMatrix55Sym covariances, const Double_t chi2);
```
Comment (shahor02) [inline DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h]: I guess here one should pass the matrices by ref. not by copy, i.e. `TrackParCovFwd(const Double_t z, const SMatrix5& parameters, const SMatrix55Sym& covariances, const Double_t chi2)` ?

File: DataFormats/Reconstruction/src/TrackFwd.cxx [hunk 3]
```diff
@@ -19,7 +19,7 @@ namespace track
 using namespace std;
 
 //_________________________________________________________________________
-TrackParCovFwd::TrackParCovFwd(const Double_t z, const SMatrix5 parameters, const SMatrix55 covariances, const Double_t chi2)
+TrackParCovFwd::TrackParCovFwd(const Double_t z, const SMatrix5 parameters, const SMatrix55Sym covariances, const Double_t chi2)
```
Comment (shahor02) [inline DataFormats/Reconstruction/src/TrackFwd.cxx]: same here, references.

**Review by shahor02 — CHANGES_REQUESTED** [review]

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h (hunk 1 above)
Comment (rpezzi) [reply DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h]: Removed,

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h (hunk 2 above)
Comment (rpezzi) [reply DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackFwd.h]: Done

**Review by shahor02 — APPROVED** [review]

**Pull request merged**